framework/TSA/AutoCorrelation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Auto Correlation Transformation for Characterization of Time-Series Data.
+"""
+import numpy as np
+
+from utils import InputData, InputTypes, xmlUtils
+from .TimeSeriesAnalyzer import TimeSeriesGenerator, TimeSeriesCharacterizer
+
+
+# utility methods
+class AutoCorrelation(TimeSeriesGenerator , TimeSeriesCharacterizer):
+  """
+    Perform AutoCorrelation Transformation on time-dependent data.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class cls.
+      @ Out, specs, InputData.ParameterInput, class to use for specifying input of cls.
+    """
+    specs = super(AutoCorrelation, cls).getInputSpecification()
+    specs.name = 'AutoCorrelation'
+    specs.description = """AutoCorrelation TimeSeriesAnalysis algorithm."""
+    specs.addSub(InputData.parameterInputFactory(
+      'nlags',
+      contentType=InputTypes.IntegerType,
+      descr=""" Number of lags to return autocorrelation for. """
+    ))
+    specs.addSub(InputData.parameterInputFactory(
+      'alpha',
```

Comment:
can we rename this to something more user friendly than algorithm friendly? Maybe "confidence" or something like that?

I don't mind calling it "alpha" internally as long as user-facing parts (input in output) use a more readable term.

Reply:
I'm down for changing the name but these argument do correspond directly to the named arguments in the underlying `statsmodels.acf` function. Does this change your perspective at all?

Reply:
I definitely am fine with using "alpha" internally; it's been our experience that way too many things have an "alpha" in the libraries we leverage, and it is frustrating from a user perspective to understand what a series of English and Greek letters mean when working in an input/output file. The description helps; if it can also be intuitive based on the name, all the better in my mind.

Reply:
I will change alpha to confidence.